tests/runner/summary/summary.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package summary
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/labstack/gommon/log"
+)
+
+func filePath(prefix, testName string) string {
+	return fmt.Sprintf("%s_summary_table_%s.json", prefix, strings.ReplaceAll(testName, string(os.PathSeparator), ""))
+}
+
+// Table is primarily used as a source of arbitrary test output. Tests can send output to the summary table and later flush them.
+// when flush is called so the table is serialized into a file that contains the test name on it, so you should use one table per test.
+// the table output is typically used as a github enhanced job summary.
+// see more: https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/
+type Table struct {
+	Test string      `json:"test"`
+	Data [][2]string `json:"data"`
+}
+
+// Output adds a pair to the table data pairs.
+func (t *Table) Output(header, value string) *Table {
+	t.Data = append(t.Data, [2]string{header, value})
```

Comment:
You should be able to write to the output file as a stream (`os.Open` creates an `io.WriteCloser`) so you don't have to buffer the data in memory. Then, `Flush()` would just close the writable stream.

Not going to block the PR on this, but if you have a chance to do that it would be very nice.

Reply:
I don't know if I understand your approach. Could you elaborate more?

I would understand if I was writing raw strings but I have to serialize to JSON first, unless you're suggesting that I use strings directly instead that would complicate things a bit

Reply:
Never mind, I missed the part where you were serializing the entire list as JSON.